app/javascript/App/bundles/NewsfeedPage/NewsList.tsx

```diff
@@ -103,18 +101,29 @@ class NewsList extends React.Component<Props, State> {
 
     const readNewsIds = JSON.parse(localStorage.getItem('readNews')) || []
 
-    const mappedItems = this.props.sortedNewsItems.map((newsItem) => {
+    const mappedItems = this.props.sortedNewsItems.map((newsItem, index) => {
       const hasRead = readNewsIds.includes(newsItem.id)
       return (
-        <NewsListItem
+        <VisibilitySensor 
           key={newsItem.id}
-          newsItem={newsItem}
-          {...this.props}
-          setActiveNewsItem={this.setActiveNewsItem}
-          // @ts-ignore FIME
-          selectCoin={(symbol) => this.selectCoin(symbol)} 
-          hasRead={hasRead}
-        />
+          onChange={(isVisible) => {
+            if (isVisible) {
+                this.props.onNewsItemShown(newsItem.id);
```

Comment:
in other parts of the app we will often short-circuit for conditional expressions 

```
isVisible && this.props.onNewsItemShown(newsItem.id)
```

Reply:
Got it! Changed to short expression